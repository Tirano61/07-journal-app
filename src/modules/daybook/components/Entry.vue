
<template>
    <div class="entry-container mb-3 pointer p-2" 
    @click="$router.push( { name:'entry', params:{id: 10} } )">
        <!-- Titulo -->
        <div class="entry-title d-flex">
            <span class="text-success fs-5 fw-bold">03</span>
            <span class="mx-1 fs-5">septiembre</span>
            <span class="mx-1 fw-light">2021, viernes</span>
        </div>

        <div class="entry-description">
            Tempor ullamco nulla minim dolor cillum laborum sit. Reprehenderit irure commodo velit dolore enim sit minim incididunt fugiat dolor in excepteur excepteur. Do sunt qui veniam exercitation sit sunt quis elit nulla consectetur laboris dolor. Esse veniam officia excepteur aute et sint labore voluptate ea. Ut quis do ut aliqua culpa aute deserunt in labore nulla amet commodo aliqua. Velit enim ad eu laborum sunt enim sint consectetur proident dolor pariatur.
        </div>
    
    </div>
</template>

<script>
export default {

}
</script>

<style lang="scss" scoped>
.entry-container{
    border-bottom: 1px solid #2c3e50;
    transition: 0.2s all ease-in;

    &:hover{
        background-color: lighten($color: grey, $amount: 45);
        transition: 0.2s all ease-in;
    }
    .entry-description{
        font-size: 12px;
    }
}

</style>