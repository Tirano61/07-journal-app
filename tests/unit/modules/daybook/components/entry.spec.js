import { shallowMount } from "@vue/test-utils";
import Entry from "../../../../../src/modules/daybook/components/Entry.vue";

import { journalState } from "../../../mock-data/test-journal-state";

describe("Pruebas en el Entry component ", () => {
    const mockRouter = {
        push: jest.fn()
    };
    const wrapper = shallowMount(Entry, {
        props: {
            entry: journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    });

    test("Debe hacer match con el snapshot", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    test("Debe redireccionar al hacer click en el entry-container", () => {

        const entryContainer = wrapper.find('.entry-container')
        entryContainer.trigger('click')
        expect(mockRouter.push).toHaveBeenCalledWith({
                name: 'entry',
                params: {
                    id: journalState.entries[0].id
                }
            }

        )
    });

    test("Pruebas en las propiedades commputadas", () => {

        expect(wrapper.vm.day).toBe(23);
        expect(wrapper.vm.month).toBe('Julio');
        expect(wrapper.vm.yearDay).toBe('Viernes');
    });
});