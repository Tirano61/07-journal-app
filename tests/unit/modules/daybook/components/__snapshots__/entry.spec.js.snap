// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Entry component  Debe hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- Titulo -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">10</span><span class="mx-1 fs-5">Septiembre</span><span class="mx-1 fw-light">Viernes</span></div>
  <div class="entry-description">Hola mundo desde mock-data</div>
</div>
`;
